sdk/core/Azure.Core.Experimental/src/DynamicJson.DynamicMetaObject.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Dynamic;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text.Json;
+
+namespace Azure.Core.Dynamic
+{
+    public partial class DynamicJson : IDynamicMetaObjectProvider
+    {
+        private static readonly MethodInfo GetPropertyMethod = typeof(DynamicJson).GetMethod(nameof(GetProperty), BindingFlags.NonPublic | BindingFlags.Instance)!;
+        private static readonly MethodInfo SetMethod = typeof(DynamicJson).GetMethod(nameof(Set), BindingFlags.NonPublic | BindingFlags.Instance, null, new Type[] { typeof(object) }, null)!;
+        private static readonly MethodInfo GetViaIndexerMethod = typeof(DynamicJson).GetMethod(nameof(GetViaIndexer), BindingFlags.NonPublic | BindingFlags.Instance)!;
+        private static readonly MethodInfo SetViaIndexerMethod = typeof(DynamicJson).GetMethod(nameof(SetViaIndexer), BindingFlags.NonPublic | BindingFlags.Instance)!;
+
+        /// <summary>
+        /// Gets a value indicating whether the current instance has a valid value of its underlying type.
+        /// </summary>
+        // TODO: Decide if this is the API we want for this.  We could also expose ValueKind.
+        public bool HasValue => _element.ValueKind != JsonValueKind.Null;
```

Comment:
@KrzysztofCwalina, curious about your perspective on this.  We could expose ValueKind instead, but anything that is a property access on DynamicJson has the following concerns:
1. Potential conflict with property in the wrapped JSON
2. Caller has to be aware of DynamicJson public API, which we're trying to avoid.

We could also remove this entirely and force callers to rely on a cast to check for null.  

What's implemented is my favorite of the options, since the usage would look most like a strongly-typed model.

Reply:
We should not expose ValueKind. ValueKind is specific to JSON and moreover to System.Text.Json. As we discussed in the past, the meta object should be independent of the format or a specific format reader technology. 

Secondly, I might not be understanding the issue (why we need HasValue at all), but I suspect it's because some JSOn properties are optional and the caller might not know if they happen to exist or not. Let me know if this is not the issue. 

If it is, I thought it would work the following way: you can cast any node of the object graph to either: 

1. A reference type, e.g. `(string)foo.Bar`, in which case the user can simply check if the reference is null.
2. A value type, e.g. `(int)foo.Bar`, in which case we would throw if Bar property does not exist.
3. A nullable variant of a value type, e.g. `(int?)foo.Bar`, in which case the user can just check HasValue on the nullable.

Reply:
I like this path forward.  We'll see it in action when I update the CLU samples -- the motivating example was here: https://github.com/Azure/azure-sdk-for-net/pull/32501/files#diff-d55b2105899a17c3fafc1a079b6bb5a777c542671abad5f96c12223977a26597R188